app/http/controllers/PreferenceController.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response, NextFunction } from "express";
+import { PreferenceService } from "../../services/PreferenceService";
+
+export const PreferenceController = {
+  /**
+   * Update the preference database
+   *
+   * @param {Request} req Express request object
+   * @param {Response} res Express response object
+   * @param {NextFunction} next Express NextFunction (used for middleware)
+   */
+  async post(req: Request, res: Response, next: NextFunction) {
+    try {
+      //@ts-ignore
+      res.send(await PreferenceService.update(req.user.uid, req.body));
```

Comment:
If you're needing to add ts-ignore to get around a typing issue, I suggest you define the type instead 👀 

That'd be in the preference service, so I'll comment there.